src/main/java/billgates/view/gui/AddEntryDialog.java

```diff
@@ -0,0 +1,111 @@
+package billgates.view.gui;
+
+import com.github.lgooddatepicker.components.DateTimePicker;
+
+import javax.swing.*;
+import java.awt.*;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+public class AddEntryDialog extends JDialog {
+
+    private int result;
+    private ZonedDateTime date;
+    private static double value;
+    private String currency;
+    private String description;
+    private String from;
+    private String to;
+    private String location;
+    private final MainFrame mainFrame;
+
+    public AddEntryDialog(MainFrame owner) {
+        super(owner);
+        this.mainFrame = owner;
+        // Create a datetime picker.
+        DateTimePicker dateTimePicker1 = new DateTimePicker();
+        add(dateTimePicker1);
+
+        JTextField valueField = new JTextField(20);
+        valueField.setDocument(new RegexDocument("\\S*"));
+        valueField.setDocument(new RegexDocument("\\w*"));
```

Comment:
Similar problem here, maybe we can only limit it to be `.` and `[0-9]`

So something like:
```suggestion
        valueField.setDocument(new RegexDocument("[0-9\\.]^*"));
```